pkg/capture/capture_manager.go

```diff
@@ -211,61 +252,92 @@ func (cm *Manager) Update(ctx context.Context, ifaces config.Ifaces) {
 		}
 	}
 
-	cm.update(ctx, ifaces, enableIfaces, disableIfaces)
+	cm.update(ctx, ifaces, enableIfaces, updateIfaces, disableIfaces)
+
+	return enableIfaces, updateIfaces, disableIfaces, nil
 }
 
-func (cm *Manager) update(ctx context.Context, ifaces config.Ifaces, enable, disable []string) {
+func (cm *Manager) update(ctx context.Context, ifaces config.Ifaces, enable, update, disable []string) {
 
 	logger, t0 := logging.FromContext(ctx), time.Now()
 
+	var disableIfaces = append(disable, update...)
+	var enableIfaces = append(enable, update...)
+
+	// execute a final writeout of all disabled interfaces in the list
+	if len(disableIfaces) > 0 {
+		cm.performWriteout(ctx, time.Now().Add(time.Second), disableIfaces...)
+	}
+
 	// To avoid any interference the update() logic is protected as a whole
 	// This also allows us to interace with the captures without copying (creating potential races)
 	cm.Lock()
 	defer cm.Unlock()
 
-	// Enable any interfaces present in the positive list
-	var rg RunGroup
-	for _, iface := range enable {
+	// store the configuration so that changes can be communicated
+	cm.lastAppliedConfig = ifaces
+
+	// Disable any interfaces present in the negative list
+	var disableRg RunGroup
+	for _, iface := range disableIfaces {
 		iface := iface
-		rg.Run(func() {
 
-			runCtx := logging.WithFields(ctx, slog.String("iface", iface))
+		mc, exists := cm.captures.Get(iface)
+		if !exists {
+			continue
+		}
+		disableRg.Run(func() {
+
+			runCtx := logging.WithFields(ctx, "iface", mc.iface)
 
-			cap := newCapture(iface, ifaces[iface]).SetSourceInitFn(cm.sourceInitFn)
-			if err := cap.run(runCtx); err != nil {
-				logging.FromContext(runCtx).Errorf("failed to start capture: %s", err)
+			logger := logging.FromContext(runCtx)
+			logger.Info("closing capture / stopping packet processing")
+
+			if err := mc.close(); err != nil {
+				logger.Errorf("failed to close capture: %s", err)
 				return
 			}
 
-			cm.captures.Set(iface, cap)
+			cm.captures.Delete(mc.iface)
 		})
 	}
-	rg.Wait()
+	disableRg.Wait()
+
+	// NOTE: make sure to use a different run group than disable's. Not doing it
```

Comment:
Weird, eh? I can't put my finger on why that happened, since line 304 should block. No matter whether we increment the wait group again.

Reply:
Well. It has nothing to do with the run group. Just ran into it again:
```
ts=2023-05-25T20:40:38.619+02:00 level=error msg="failed to free capture resources: cannot call Free() on open capture source, call Close() first" version=devel iface=eth1
```
Call that lead to it:
```
gpctl -s localhost:8145 config -f dummy.conf
```
Config that goprobe was started with:
```
"interfaces" : {
      "eth0" : {
         "ring_buffer_block_size" : 10485760,
         "ring_buffer_num_blocks" : 2,
         "promisc" : true
      },
      "eth1" : {
         "ring_buffer_block_size" : 10485760,
         "ring_buffer_num_blocks" : 2,
         "promisc" : true
      }
   }
```

File contents of `dummy.conf`
```
interfaces:
  eth0:
    ring_buffer_block_size: 10485760
    ring_buffer_num_blocks: 4
    promisc: true
```

Reply:
@fako1024 : let's have a closer look at this one. It might be that we missed something here. Because this PR is the first time that we shut down an interface while goProbe keeps running.

Reply:
Yeah, I saw that once, too when I was running a test instance locally yesterday. I'll raise an extra issue, this has indeed nothing to do with the `RunGroup` / shadowing and needs to be looked at.